src/v/cluster/commands.h

```diff
@@ -264,6 +270,30 @@ using update_user_cmd = controller_command<
   model::record_batch_type::user_management_cmd,
   serde_opts::adl_and_serde>;
 
+using create_role_cmd = controller_command<
+  int8_t, // unused
+  upsert_role_cmd_data,
+  create_role_cmd_type,
+  model::record_batch_type::role_management_cmd>;
+
+using delete_role_cmd = controller_command<
+  int8_t, // unused
+  security::role_name,
```

Comment:
Let's also wrap this in a struct, for consistency, and you never know if we want to implement soft deletes or something and need to add more info.

Reply:
Ya, I saw some "bare" named_type<string>s in other commands, but great point wrt soft deletion.

Reply:
done